### PR TITLE
[Merged by Bors] - refactor(LinearAlgebra/BilinearMap): Left composition, bilinear over different rings

### PR DESCRIPTION
**Pull request by mans0954**
Generalise `compl₂` and `compl₁₂` for left composition with maps which are linear over different rings in the first and second variable.

Needed for #9334

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mans0954** [comment]
!bench

**Comment by mans0954** [comment]
@eric-wieser do you have a moment to review this PR please? It's quite a small one.

File: Mathlib/LinearAlgebra/BilinearMap.lean [hunk 1]
```diff
@@ -325,8 +325,14 @@ end
 
 /-- Composing a linear map `Q → N` and a bilinear map `M → N → P` to
 form a bilinear map `M → Q → P`. -/
-def compl₂ (g : Q →ₛₗ[σ₄₂] N) : M →ₛₗ[σ₁₃] Q →ₛₗ[σ₄₃] P :=
-  (lcompₛₗ _ _ g).comp f
+def compl₂ {R₅ : Type*} [CommSemiring R₅] [Module R₅ P] [SMulCommClass R₃ R₅ P] {σ₁₅ : R →+* R₅}
+    (h: M →ₛₗ[σ₁₅] N →ₛₗ[σ₂₃] P) (g : Q →ₛₗ[σ₄₂] N) : M →ₛₗ[σ₁₅] Q →ₛₗ[σ₄₃] P where
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/BilinearMap.lean:329]:
```suggestion
    (h : M →ₛₗ[σ₁₅] N →ₛₗ[σ₂₃] P) (g : Q →ₛₗ[σ₄₂] N) : M →ₛₗ[σ₁₅] Q →ₛₗ[σ₄₃] P where
```

File: Mathlib/LinearAlgebra/BilinearMap.lean [hunk 2]
```diff
@@ -325,8 +325,14 @@ end
...
+  toFun a := (lcompₛₗ P σ₂₃ g) (h a)
+  map_add' _ _ := by
+    simp  [map_add]
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/BilinearMap.lean:332]:
```suggestion
    simp [map_add]
```

**Review by eric-wieser — APPROVED** [review]
bors d+

The variables are getting a bit unmanageable here, but I don't have any suggestions to clean them up. Just some whitespace suggestions.

**Comment by eric-wieser** [comment]
#26458 takes this generalization further.

**Pull request closed without merging**